packages/amplify-category-analytics/src/utils/pinpoint-helper.ts

```diff
@@ -59,3 +63,22 @@ export const hasResource = (context: $TSContext): boolean => {
 
   return pinpointApp !== undefined;
 };
+
+/**
+ * Checks if pinpoint has in app messaging policy
+ */
+export const pinpointHasInAppMessagingPolicy = (context: $TSContext): boolean => {
+  const resources = getAnalyticsResources(context, AmplifySupportedService.PINPOINT);
+  if (resources?.length > 0) {
+    const pinpointCloudFormationTemplatePath = path.join(
+      pathManager.getBackendDirPath(),
+      AmplifyCategories.ANALYTICS,
+      resources[0].resourceName,
+      `pinpoint-cloudformation-template.json`,
+    );
+    const { cfnTemplate } = readCFNTemplate(pinpointCloudFormationTemplatePath, { throwIfNotExist: false }) || {};
+    return !!cfnTemplate?.Parameters?.pinpointInAppMessagingPolicyName;
```

Comment:
I'd recommend putting `pinpointInAppMessagingPolicyName` in a const that is shared between where this value is written and read to ensure it stays in sync

Reply:
updated